storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/SnowflakeQueryTemplater.java

```diff
@@ -70,30 +70,34 @@ public SnowflakeQueryTemplater(JdbcConnectionProvider connectionProvider) {
     return createEntityTableRowCountQueries;
   }
 
+  // TODO: support filepath as s3 location
   @Override
   protected List<String> createLoadEntityQuery(
-      String destinationTable, String temporaryTable, File filePath) {
+      String destinationTable, String stagingLocation, URI fileUri) {
     List<String> queries = new ArrayList<>();
-    queries.add(
-        String.format("CREATE TABLE %s AS (SELECT * FROM %s);", temporaryTable, destinationTable));
-    queries.add(
+    String csvFormatQuey =
         String.format(
-            "create or replace file format CSV_format type = 'CSV' field_delimiter = ',' skip_header=1;"));
-    queries.add(String.format("create or replace stage my_stage file_format = CSV_format;"));
-    queries.add(String.format("put file://%s @my_stage auto_compress=false;", filePath));
-    String fileName = filePath.getName();
-    queries.add(
+            "create or replace file format CSV_format type = 'CSV' field_delimiter = ',' skip_header=1;");
+    String createStageQuery =
         String.format(
-            "COPY INTO %s FROM '@my_stage/%s' FILE_FORMAT = CSV_format on_error = 'skip_file';",
-            temporaryTable, fileName));
-    queries.add(
-        String.format("INSERT INTO %s SELECT * FROM %s;", destinationTable, temporaryTable));
+            "create or replace stage my_s3_stage\n"
+                + "  storage_integration = s3_int\n"
+                + "  url = '%s';",
+            stagingLocation);
+    // TODO: fileUri must be the in the stagingLocation??
+    String folderPath = fileUri.toString().substring(stagingLocation.length());
```

Comment:
there could be a better way to get the folder path, in order to avoid using` toString()`?

Reply:
that entity source file URIs shouldn't be coupled to the staging URI.  They should all be full complete URIs that should be used independently.

Reply:
Since we have a list of entity source file, I'm wondering shall we provide a list of storage_integration? I'm not sure what is the best way to make storage integration configurable.

Reply:
no I think just one integration for both staging and sources for now

Reply:
ok, do you think let the user set the storage_integration in the `config` would be a good approach?

Reply:
i'm talking about in  JdbcHistoricalRetriever(Map<String, String> config, JdbcQueryTemplater queryTemplater)